plugins/modules/files/sapcar_extract.py

```diff
@@ -0,0 +1,220 @@
+    # Iterate over all the entries
+    for entry in listOfFile:
+        # Create full path
+        fullPath = os.path.join(dir_name, entry)
+        # If entry is a directory then get the list of files in this directory
+        if os.path.isdir(fullPath):
+            allFiles = allFiles + [fullPath]
+            allFiles = allFiles + get_list_of_files(fullPath)
+        else:
+            allFiles.append(fullPath)
+    return allFiles
+
+
+def download_SAPCAR(binary_path):
+    bin_path = None
+    is_random = False
+    random_file = NamedTemporaryFile(delete=False)
+    # download sapcar binary if url is provided otherwise path is returned
+    if binary_path is not None:
+        if binary_path.startswith('http://' or 'https://'):
+            with open_url(binary_path) as response:
+                with open(random_file, 'wb') as out_file:
+                    data = response.read()
+                    out_file.write(data)
+            os.chmod(random_file, 0o700)
+            bin_path = random_file
+            is_random = True
+        else:
+            bin_path = binary_path
+    return [bin_path, is_random]
+
+
+def check_if_Present(command, path, dest, signature, manifest):
+    # manipuliating output from SAR file for compare with already extracted files
+    iter_command = [command, '-tvf', path]
+    sar_out = module.run_command(iter_command)[1]
+    sar_raw = sar_out.split("\n")[1:]
+    if dest[-1] != "/":
+        dest = dest + "/"
+    sar_files = [dest + x.split(" ")[-1] for x in sar_raw if x]
+    # remove any SIGNATURE.SMF from list because it will not unpacked if signature is false
+    if not signature:
+        sar_files = [item for item in sar_files if '.SMF' not in item]
+    # if signature is renamed manipulate files in list of sar file for compare.
+    if manifest != "SIGNATURE.SMF":
+        sar_files = [item for item in sar_files if '.SMF' not in item]
+        sar_files = sar_files + [manifest]
+    # get extracted files if present
+    files_extracted = get_list_of_files(dest)
+    # compare extracted files with files in sar file
+    present = all(elem in files_extracted for elem in sar_files)
+    return present
+
+
+def main():
+    global module
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(type='path', required=True),
+            dest=dict(type='path'),
+            binary_path=dict(type='path'),
+            signature=dict(type='bool', default=False),
+            manifest=dict(type='str', default="SIGNATURE.SMF"),
+            remove=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+    )
+    rc, out, err = [0, "", ""]
+    params = module.params
+    check_mode = module.check_mode
+
+    path = params['path']
+    dest = params['dest']
+    signature = params['signature']
+    manifest = params['manifest']
+    remove = params['remove']
+
+    bin_path_raw = download_SAPCAR(params['binary_path'])
+    bin_path = bin_path_raw[0]
```

Comment:
Use a tuple assignment instead:
```suggestion
    bin_path, bin_path_cleanup = download_SAPCAR(params['binary_path'])
```